File: CS/CustomDraw/Form1.cs
using DevExpress.Pdf;
using System;
using System.Drawing;
using System.Windows.Forms;

namespace CustomDraw {
    public partial class Form1 : Form {

        public Form1() {
            InitializeComponent();
            pdfViewer1.MouseDown += pdfViewer1_MouseDown;
            pdfViewer1.MouseMove += pdfViewer1_MouseMove;
            pdfViewer1.MouseUp += pdfViewer1_MouseUp;
            pdfViewer1.Paint += pdfViewer1_Paint;
        }

        bool mouseButtonPressed = false;
        PdfDocumentPosition startPosition;
        PdfDocumentPosition endPosition;

        void pdfViewer1_MouseDown(object sender, MouseEventArgs e) {
            if (e.Button == MouseButtons.Left) {
                startPosition = pdfViewer1.GetDocumentPosition(e.Location);
                endPosition = null;
                mouseButtonPressed = true;
                pdfViewer1.Invalidate();
            }
        }

        void pdfViewer1_MouseMove(object sender, MouseEventArgs e) {
            if (mouseButtonPressed) {
                endPosition = pdfViewer1.GetDocumentPosition(e.Location);
                pdfViewer1.Invalidate();
            }
        }

        void pdfViewer1_MouseUp(object sender, MouseEventArgs e) {
            mouseButtonPressed = false;
        }

        void pdfViewer1_Paint(object sender, PaintEventArgs e) {
            Graphics g = e.Graphics;
            g.DrawRectangle(Pens.Red, new Rectangle(150, 150, 800, 50));

            if (startPosition != null && endPosition != null) {
                PointF startPoint = pdfViewer1.GetClientPoint(startPosition);
                PointF endPoint = pdfViewer1.GetClientPoint(endPosition);

                g.FillRectangle(new SolidBrush(Color.FromArgb(128, Color.Aqua)),
                    RectangleF.FromLTRB(Math.Min(startPoint.X, endPoint.X), Math.Min(startPoint.Y, endPoint.Y),
                    Math.Max(startPoint.X, endPoint.X), Math.Max(startPoint.Y, endPoint.Y)));
            }
        }
    }
}

File: VB/CustomDraw/Form1.vb
Imports DevExpress.Pdf
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Namespace CustomDraw

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            AddHandler pdfViewer1.MouseDown, AddressOf pdfViewer1_MouseDown
            AddHandler pdfViewer1.MouseMove, AddressOf pdfViewer1_MouseMove
            AddHandler pdfViewer1.MouseUp, AddressOf pdfViewer1_MouseUp
            AddHandler pdfViewer1.Paint, AddressOf pdfViewer1_Paint
        End Sub

        Private mouseButtonPressed As Boolean = False

        Private startPosition As PdfDocumentPosition

        Private endPosition As PdfDocumentPosition

        Private Sub pdfViewer1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            If e.Button = MouseButtons.Left Then
                startPosition = pdfViewer1.GetDocumentPosition(e.Location)
                endPosition = Nothing
                mouseButtonPressed = True
                pdfViewer1.Invalidate()
            End If
        End Sub

        Private Sub pdfViewer1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            If mouseButtonPressed Then
                endPosition = pdfViewer1.GetDocumentPosition(e.Location)
                pdfViewer1.Invalidate()
            End If
        End Sub

        Private Sub pdfViewer1_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs)
            mouseButtonPressed = False
        End Sub

        Private Sub pdfViewer1_Paint(ByVal sender As Object, ByVal e As PaintEventArgs)
            Dim g As Graphics = e.Graphics
            g.DrawRectangle(Pens.Red, New Rectangle(150, 150, 800, 50))
            If startPosition IsNot Nothing AndAlso endPosition IsNot Nothing Then
                Dim startPoint As PointF = pdfViewer1.GetClientPoint(startPosition)
                Dim endPoint As PointF = pdfViewer1.GetClientPoint(endPosition)
                Using blueBrush As SolidBrush = New SolidBrush(Color.FromArgb(128, Color.Aqua))
                    g.FillRectangle(blueBrush, RectangleF.FromLTRB(Math.Min(startPoint.X, endPoint.X), Math.Min(startPoint.Y, endPoint.Y), Math.Max(startPoint.X, endPoint.X), Math.Max(startPoint.Y, endPoint.Y)))
                End Using
            End If
        End Sub
    End Class
End Namespace
